### PR TITLE
Moved Settings to be a regular page, not a free-floating window

### PR DESCRIPTION
**Pull request by andreww-msft**
## Summary
Before this, PI Settings were provided in a separate free-floating window. This change moves PI Settings into the main window as a page, using the same basic navigation model as Dev Home Settings.

## Details
This meant creating a new viewmodel and page for the main settings UI, and for each of the 4 sub-pages, and registering these with the navigation service and page service. Per Dev Home model, moved the various UI event handlers to command-binding with RelayCommands in the viewmodel (although I mostly left the AdditionalTools page alone after the port, because we have a backlog item to redesign this piece anyway).

Reimplemented all the Settings sub-pages to match Dev Home's model: moved custom border items to SettingsCards, SettingsExpanders and so on. Added simple breadcrumb navigation. Created PIAppInfoService, which is a subset of Dev Home's AppInfoService. Deleted the old SettingsToolWindow & ToolWindow classes, moved the AddTools and EditTools controls to the Controls folder/namespace and deleted the SettingsUi folder.

In the ExpandedView, moved the Settings button to the bottom of the left nav-list, and reworked the grid layout to ensure Settings is at the bottom regardless of whether there's a perf counter row. Deleted the SettingsNavLink class as its now redundant. Note that the main PI nav mechanism is different from Dev Home's main nav model, and we have a backlog item to update this to a NavigationView.

Here's the main Settings page in the PI expanded view:

PI-settings-main

Breadcrumbs and page navigation...

PI-settings-breadcrumbs

### CODE REVIEW

File: tools/PI/DevHome.PI/PIApp.xaml [hunk 1]
```diff
@@ -2,7 +2,9 @@
 <Application
     x:Class="DevHome.PI.App"
     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
-    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:i="using:Microsoft.Xaml.Interactivity"
+    xmlns:behaviors="using:DevHome.Common.Behaviors">
```
Comment (zadesai) [inline tools/PI/DevHome.PI/PIApp.xaml:7]: Curious, why do we need these?

File: tools/PI/DevHome.PI/ViewModels/SettingViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.PI.Models;
+
+namespace DevHome.PI.ViewModels;
+
+public partial class SettingViewModel : ObservableObject
```
Comment (zadesai) [inline tools/PI/DevHome.PI/ViewModels/SettingViewModel.cs:10]: >SettingViewModel [](http://example.com/codeflow?start=21&length=16)

nit: typo, we probably want it as SettingsViewModel.

File: tools/PI/DevHome.PI/PIApp.xaml (hunk 1 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/PIApp.xaml]: Leftover from when I had a more complex breadcrumb style. Removed in 2nd commit.

File: tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs [hunk 3]
```diff
@@ -234,4 +239,27 @@ public void Navigate()
         var navigationService = Application.Current.GetService<INavigationService>();
         navigationService.NavigateTo(Links[SelectedNavLinkIndex]?.PageViewModel?.FullName!);
     }
+
+    public void NavigateToSettings(string viewModelType)
+    {
+        // Because the Settings item isn't part of our NavLink list, when the user selects Settings,
+        // we need to move the list selection so that when they subsequently select an item from
+        // the NavLinks, we'll navigate to the correct page even if that was the previously-selected item.
```
Comment (zadesai) [inline tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs:247]: The screenshot in the PR shows Insights page selected when Settings is selected. Is it possible to not have a selected page?

File: tools/PI/DevHome.PI/ViewModels/SettingsPageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    public ObservableCollection<Breadcrumb> Breadcrumbs { get; }
+
+    [ObservableProperty]
+    private ObservableCollection<SettingViewModel> _settingsList = new();
+
+    public SettingsPageViewModel()
+    {
+        var settings = new[]
+        {
+            new Setting("Preferences", CommonHelper.GetLocalizedString("SettingsPreferencesHeader"), CommonHelper.GetLocalizedString("SettingsPreferencesSubHeader"), "\uE713"),
```
[the remark this answers is not in the document]
Comment (andreww-msft) [reply tools/PI/DevHome.PI/ViewModels/SettingsPageViewModel.cs:29]: Hmm... we use this pattern throughout, as does main Dev Home, so if we're going to change it we'd want to change it everywhere...

File: tools/PI/DevHome.PI/ViewModels/SettingViewModel.cs (hunk 2 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/ViewModels/SettingViewModel.cs:10]: SettingsPageViewModel is the main page for all settings. SettingViewModel is for one setting each time.

File: tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs (hunk 3 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs:247]: Yes, unfortunately this is an artifact of how our main nav list doesn't use a NavigationView. We have a backlog item to fix this all-up.

File: tools/PI/DevHome.PI/ViewModels/SettingViewModel.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (andreww-msft) [reply tools/PI/DevHome.PI/ViewModels/SettingViewModel.cs:10]: As above, SettingViewModel is for one setting, SettingsPageViewModel is for the whole collection of settings.

File: tools/PI/DevHome.PI/ViewModels/SettingsPageViewModel.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (zadesai) [reply tools/PI/DevHome.PI/ViewModels/SettingsPageViewModel.cs:29]: Interesting, I had made changes to have a const defined and moved away from this pattern in PI. But if we follow the same in Dev Home, I guess its ok.

File: tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs (hunk 3 above)
Comment (zadesai) [reply tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs:247]: sounds good

**Review by zadesai — APPROVED** [review]
:shipit:

**Review by jaholme — APPROVED** [review]
:shipit:

**Pull request merged**